server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -194,12 +205,19 @@ macro_rules! record_stream_to_response_stream {
                                     CompressionMethod::Zstd => arrow_payload::Compression::Zstd,
                                 };
 
-                                ReadResponse {
-                                    header: Some(error::build_ok_header()),
-                                    output: Some(Arrow(ArrowPayload {
+                                let output = match schema.column_by_name(METRICS) {
```

Comment:
It's weird to see the `METRICS` occurs in the schema. Maybe we can define an enum including RecordBatch and Metrics, as the record_stream's element.